pkg/systemd/quadlet/quadlet.go

```diff
@@ -1321,12 +1367,25 @@ func ConvertImage(image *parser.UnitFile) (*parser.UnitFile, string, error) {
 		imageName = name
 	}
 
-	return service, imageName, nil
+	// Store the name of the created resource
+	unitInfo.ResourceName = imageName
+
+	return service, nil
 }
 
-func ConvertBuild(build *parser.UnitFile, names map[string]string) (*parser.UnitFile, string, error) {
+func ConvertBuild(build *parser.UnitFile, unitsInfoMap map[string]*UnitInfo) (*parser.UnitFile, error) {
+	unitInfo, ok := unitsInfoMap[build.Filename]
+	if !ok {
+		return nil, fmt.Errorf("internal error while processing network %s", build.Filename)
+	}
+
+	// Fast fail is ResouceName is not set
+	if len(unitInfo.ResourceName) == 0 {
+		return nil, fmt.Errorf("no ImageTag key specified")
+	}
+
 	service := build.Dup()
-	service.Filename = replaceExtension(build.Filename, ".service", "", "-build")
+	service.Filename = fmt.Sprintf("%s.service", unitInfo.ServiceName)
```

Comment:
One enhancement would be to add a function unitInfo.Filename() and clean this up.

service.Filename=unitInfo.Filename()

Since this is replicated everywhere.

Reply:
Done